Adding ability to list all resources in a given environment with rad resource list

willvelida opened the pull request:
# Description

This pull request enhances the functionality of `rad resource list` command by adding support for the environment flag. With this change, users can now list all resources of a specified type in an environment using the `--environment` flag or its shorthand form `-e`.

The following additions were made:

- Added the environment flag to the command definition
- Updated the validation logic to handle environment name from command line or workspace
- Enhanced the Run command to support listing resources in an environment
- Added capability to list all resources in an environment with no resource type specified
- Updated command examples to demonstrate the new functionality

## Type of change

- This pull request adds or changes features of Radius and has an approved issue. Fixes: #7789 

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

<!--
This checklist uses "TaskRadio" comments to make certain options mutually exclusive.
See: https://github.com/mheap/require-checklist-action?tab=readme-ov-file#radio-groups
For details on how this works and why it's required.
-->

- An overview of proposed schema changes is included in a linked GitHub issue.
    - [x] Yes <!-- TaskRadio schema -->
    - [ ] Not applicable <!-- TaskRadio schema -->
- A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
    - [ ] Yes <!-- TaskRadio design-pr -->
    - [x] Not applicable <!-- TaskRadio design-pr -->
- The design document has been reviewed and approved by Radius maintainers/approvers.
    - [ ] Yes <!-- TaskRadio design-review -->
    - [x] Not applicable <!-- TaskRadio design-review -->
- A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
    - [ ] Yes <!-- TaskRadio samples-pr -->
    - [x] Not applicable <!-- TaskRadio samples-pr -->
- A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
    - [ ] Yes <!-- TaskRadio docs-pr -->
    - [x] Not applicable <!-- TaskRadio docs-pr -->
- A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.
    - [ ] Yes <!-- TaskRadio recipes-pr -->
    - [x] Not applicable <!-- TaskRadio recipes-pr -->

willvelida committed:
Adding ability to list all resources in a given environment with rad resource list. Fixes #7789

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

ytimocin commented on pkg/cli/cmd/resource/list/list.go:
```diff
@@ -148,25 +180,57 @@ func (r *Runner) Run(ctx context.Context) error {
-		resourceList, err = client.ListResourcesOfType(ctx, r.ResourceType)
+	// If no resource type is specified, but an environment name is provided,
+	// list all resources in the environment
+	if r.ResourceType == "" && r.EnvironmentName != "" {
```
What if both environment and application names are provided? Should we filter resources for both? cc/ @willtsai

willtsai replied:
yes, if both are provided, then filter for both.

willvelida replied:
@ytimocin @willtsai - I've committed a change to filter when both environment and application names are provided, as well as adding new test cases. Let me know if I'm on the right track here :)

ytimocin commented on pkg/cli/cmd/resource/list/list.go:
```diff
@@ -58,15 +58,28 @@ rad resource list Applications.Core/containers --application icecream-store
+rad resource list --environment dev-env
+
+# list all resources in the default environment
+rad resource list
```
Would it make sense to return all resources in all environments if this is the case? This is not in the scope of this PR but just wanted to brainstorm on this potential feature. cc/ @willtsai

willtsai replied:
no, I think that listing the resources in the default environment is good since using the default if none provided is a pattern that's applied across the other commands.

willtsai replied:
Is it possible to display the name of the environment when rendering the results? This would be especially useful in the case where we fall back to a default when user doesn't provide an environment flag

willvelida replied:
@willtsai - Is this something you want implemented in this PR? If so, am I right in saying that the changes would need to be made in `objectformats`? @ytimocin

willtsai replied:
Yes, if possible it would be great to include it in this PR. I'll get @ytimocin to confirm for you.

willvelida replied:
@willtsai & @ytimocin  - I've pushed a change to display the environment name when rendering `rad resource list` - Please let me know if I've got it right or need to make changes :)

ytimocin commented on pkg/cli/cmd/resource/list/list_test.go:
```diff
@@ -262,4 +280,124 @@ func Test_Run(t *testing.T) {
 			require.Equal(t, expected, outputSink.Writes)
 		})
 	})
+	t.Run("List resources by type in environment", func(t *testing.T) {
```
nit: extra line before this one

willvelida replied:
Done 😃

ytimocin commented on pkg/cli/cmd/resource/list/list_test.go:
```diff
@@ -61,6 +61,24 @@ func Test_Validate(t *testing.T) {
 		},
+		{
+			Name:          "Valid List Command with environment",
+			Input:         []string{"Applications.Core/containers", "-e", "test-env"},
```
Should we check `--environment` just in case?

willvelida replied:
@ytimocin - I've added a couple of test cases to include the full `--environment` flag 😃

ytimocin reviewed (commented):
Looks great so far! Thank you for this contribution!

willvelida committed:
Merge branch 'radius-project:main' into feature/7789

willvelida committed:
Adding full environment flag in list_test.go test cases

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Filtering resources when application and environment names are provided in rad resource list. Adding test cases and reorganizing code

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Merge branch 'main' into feature/7789

brooke-hamilton reviewed (changes requested):
🥇 Thank you for this PR! 🥇 

I did a local build of the branch and tested a deployment of the default application that is generated with `rad init`. I see some possibly unintended differences in the behavior of the `rad resource list` command. It's possible that I misconfigured something - so please let me know if you are seeing different results.

The first difference is that current release version of `rad resource list` returns an error if I run it without any parameters. It looks like this, which is the expected result.

```shell
$ rad version
CLI Version Information:
RELEASE   VERSION   BICEP     COMMIT
0.47.0    v0.47.0   0.35.1    8b18539f5b221e10818d334c68b6763b460267c5

Control Plane Information:
STATUS     VERSION
Installed  0.47.0

~ $ rad resource list
Error: accepts 1 arg(s), received 0

TraceId:  2f06d9310027af31438ceac24c8e873f
```
The above error is expected because the command requires an argument.

If I supply a valid parameter, without any flags, I get the expected result.

```shell
$ rad resource list Applications.Core/containers
RESOURCE  TYPE                          GROUP     STATE
demo      Applications.Core/containers  default   Succeeded
```

When I run the Radius CLI built from the branch in this PR, I get a different result: no error when I would expect an error.

```shell
$ rad version
CLI Version Information:
RELEASE   VERSION              BICEP     COMMIT
edge      v0.46.0-28-g681a93e  0.35.1    681a93e72e10e11ed589d372a7fb26f8649bd169

Control Plane Information:
STATUS     VERSION
Installed  0.47.0

$ rad resource list
RESOURCE  TYPE      GROUP     STATE
```

A second issue is that when I run `rad resource list` with a valid resource type name, I get no results. And when I supply the environment name argument, I also get no results.

```shell
$ rad resource list Applications.Core/containers
RESOURCE  TYPE      GROUP     STATE
```

Supplying the environment name does not resolve the issue:

```shell
$ rad env list
RESOURCE  TYPE                            GROUP     STATE
default   Applications.Core/environments  default   Succeeded

$ rad resource list Applications.Core/containers --environment default
RESOURCE  TYPE      GROUP     STATE
```

willvelida committed:
fix(cli): enhance rad resource list command for better filtering

1. Changed  to  to require resource type argument
2. Enhanced  function to properly handle environment resource filtering:
   - Added special case for environment resources themselves
   - Implemented case-insensitive comparison of environment IDs
   - Added support for matching both short names and fully-qualified IDs
3. Added comprehensive unit tests for  function
4. Updated existing tests to match new behavior

This fixes issues with the  flag not properly returning resources when the
environment names are specified in different formats or cases.

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Fixing environment filtering in rad resource list. Adding tests

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida commented:
@brooke-hamilton - I've done some testing on my side, and I've found the following:

Issue 1 - ```rad resource list``` not returning error:

I verified that I was also seeing this, so I've committed a change that ensures the error prints as expected:

```bash
vscode ➜ /workspaces/radius (feature/7789) ./dist/linux_arm64/release/rad resource list
Error: accepts 1 arg(s), received 0

TraceId:  5db66bef4c6874567694f8ea51181381
```

Issue 2: Supplying a valid resource type with ```rad resource list`` and getting no results.

I have also ensured that when we supply a valid parameter without any flags, we get the expected result like so:

```bash
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers
RESOURCE  TYPE                          GROUP     STATE
demo      Applications.Core/containers  default   Succeeded
```

Issue 3 - Supplying environment name (with long and shorthand flag) returns no results.


I have also made sure that when we provide a valid type with the environment flag, that we get the expected results like so:

```bash
vscode ➜ /workspaces/radius (feature/7789) $./dist/linux_arm64/release/rad resource list Applications.Core/containers --environment default
RESOURCE  TYPE                          GROUP     STATE
demo      Applications.Core/containers  default   Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -e default
RESOURCE  TYPE                          GROUP     STATE
demo      Applications.Core/containers  default   Succeeded
```

Please let me know if this:

- Meets the acceptance criteria (and what you'd expect to see)
- Runs as I've described in your environment.

If you need me to change anything, please let me know 🙂

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Moving if condition in isResourceInEnvironment method in management.go and implementing helper method

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida commented on pkg/cli/clients/management.go:
```diff
@@ -1022,9 +1022,48 @@ func isResourceInApplication(resource generated.GenericResource, applicationID s
 	if !found {
+		// Special case for environment resources themselves
+		if resource.Type != nil && strings.EqualFold(*resource.Type, "Applications.Core/environments") {
+			if resource.ID != nil {
```
@ytimocin - Done

willvelida commented on pkg/cli/clients/management.go:
```diff
@@ -1033,10 +1072,29 @@ func isResourceInEnvironment(resource generated.GenericResource, environmentID s
+	if shortNameEnv != "" && shortNameAssociated != "" &&
+		strings.EqualFold(shortNameEnv, shortNameAssociated) {
+		return true
+	}
```
@ytimocin - Also done

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Adding ability to display environment when rendering rad resource list. Updating tests

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Merge branch 'main' into feature/7789

brooke-hamilton commented on pkg/cli/clients/management_test.go:
```diff
@@ -1355,3 +1353,158 @@ func setCapture(ctx context.Context, response *http.Response) {
+		expected      bool
+	}
+
+	fullEnvID := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Applications.Core/environments/env1"
```
Is this a valid environment ID? This looks more like the provider scope to me, but maybe there's a scenario in which this becomes an ID?

willvelida replied:
Fixed

brooke-hamilton commented on pkg/cli/clients/management_test.go:
```diff
@@ -1355,3 +1353,158 @@ func setCapture(ctx context.Context, response *http.Response) {
+	}
+
+	fullEnvID := "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Applications.Core/environments/env1"
+	shortEnvID := "env1"
```
In what scenario would the ID be represented as this short form instead of the full ID? Looking at json output from a valid environment, it usually looks like this: `"/planes/radius/local/resourcegroups/default/providers/Applications.Core/environments/default"`

brooke-hamilton commented on pkg/cli/clients/management.go:
```diff
@@ -1022,9 +1022,46 @@ func isResourceInApplication(resource generated.GenericResource, applicationID s
+		}
+
+		// Special handling for the default environment - resources without an explicit
+		// environment property are considered to be in the default environment
```
Is it possible for a resource to not have an environment?

willvelida replied:
Based off the code in main, I believe so: https://github.com/radius-project/radius/blob/main/pkg/cli/clients/management.go#L1026

brooke-hamilton reviewed (changes requested):
When I run `make test` I get these test errors:

```shell
=== Failed
=== FAIL: pkg/cli/cmd/resource/list  (0.00s)
FAIL    github.com/radius-project/radius/pkg/cli/cmd/resource/list [build failed]

=== Errors
pkg/cli/cmd/resource/list/list_test.go:336:21: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal

pkg/cli/cmd/resource/list/list_test.go:455:21: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal

pkg/cli/cmd/resource/list/list_test.go:546:20: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal

pkg/cli/cmd/resource/list/list_test.go:608:20: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal


DONE 3743 tests, 3 skipped, 1 failure, 4 errors in 246.541s
make: *** [build/test.mk:47: test] Error 1
```

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Fixing environment ID formats in management_test.go. Fixing tests in list_test.go. Introducing getResourceShortName helper function in management.go

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida commented on pkg/cli/clients/management.go:
```diff
@@ -1033,10 +1070,29 @@ func isResourceInEnvironment(resource generated.GenericResource, environmentID s
+	if strings.Contains(environmentID, "/") {
+		parts := strings.Split(environmentID, "/")
+		shortNameEnv = parts[len(parts)-1]
+	}
```
Introduced helper function for logic

willvelida committed:
Merge branch 'main' into feature/7789

willvelida commented:
> When I run `make test` I get these test errors:
> 
> ```shell
> === Failed
> === FAIL: pkg/cli/cmd/resource/list  (0.00s)
> FAIL    github.com/radius-project/radius/pkg/cli/cmd/resource/list [build failed]
> 
> === Errors
> pkg/cli/cmd/resource/list/list_test.go:336:21: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal
> 
> pkg/cli/cmd/resource/list/list_test.go:455:21: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal
> 
> pkg/cli/cmd/resource/list/list_test.go:546:20: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal
> 
> pkg/cli/cmd/resource/list/list_test.go:608:20: cannot use map[string]map[string]any{…} (value of type map[string]map[string]any) as map[string]*"github.com/radius-project/radius/pkg/ucp/api/v20231001preview".ResourceTypeSummaryResultAPIVersion value in struct literal
> 
> 
> DONE 3743 tests, 3 skipped, 1 failure, 4 errors in 246.541s
> make: *** [build/test.mk:47: test] Error 1
> ```

Fixed the tests now

willvelida committed:
Merge branch 'main' into feature/7789

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Fixing conflicts manually, and incorporating changes for UCPClientFactory

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

brooke-hamilton reviewed (changes requested):
Thanks for continuing to work with us on this.

I ran into an issue - it could be the way I set things up, and it could be that I'm testing the wrong scenario, so please let me know if there are setup or test steps I am missing.

### Setup
I deployed the same app to two different group/environment combinations. The overall steps are below. I put links to the scripts I used for reference.

1. Checkout the PR and run `make install` to install the `rad` CLI.
1. [Create new cluster ](https://github.com/brooke-hamilton/dotfiles/blob/main/radius/setup_k3d.sh)(I used k3d in the dev container).
1. [Deploy](https://github.com/brooke-hamilton/dotfiles/blob/main/radius/test/resource-list-test.sh) the same app to two different group/environments.
1. Run `rad resource list Applications.Core/containers -a todoapp -e peachenv`

### Issue

I'm able to see the resource if I filter on group, which is expected.
```shell
$ rad resource list Applications.Core/containers -a todoapp -g peachgroup
RESOURCE       TYPE                          GROUP       ENVIRONMENT  STATE
todocontainer  Applications.Core/containers  peachgroup  default      Succeeded
```

If I add environment I cannot see the resource.
```shell
$ rad resource list Applications.Core/containers -a todoapp -e peachenv -g peachgroup
RESOURCE  TYPE      GROUP     ENVIRONMENT  STATE
```

If I filter on environment only, I cannot see the resources.
The result looks like this:
```shell
$ rad resource list Applications.Core/containers -a todoapp -e peachenv
The application "todoapp" could not be found in workspace "k3d". Make sure you specify the correct application with '-a/--application'
```

willvelida committed:
Fixing environment displays and filtering logic in rad resource list

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida committed:
Fixing formatting in clients.go and management.go

Signed-off-by: Will Velida <willvelida@hotmail.co.uk>

willvelida commented:
> Thanks for continuing to work with us on this.
> 
> I ran into an issue - it could be the way I set things up, and it could be that I'm testing the wrong scenario, so please let me know if there are setup or test steps I am missing.
> 
> ### Setup
> I deployed the same app to two different group/environment combinations. The overall steps are below. I put links to the scripts I used for reference.
> 
> 1. Checkout the PR and run `make install` to install the `rad` CLI.
> 2. [Create new cluster ](https://github.com/brooke-hamilton/dotfiles/blob/main/radius/setup_k3d.sh)(I used k3d in the dev container).
> 3. [Deploy](https://github.com/brooke-hamilton/dotfiles/blob/main/radius/test/resource-list-test.sh) the same app to two different group/environments.
> 4. Run `rad resource list Applications.Core/containers -a todoapp -e peachenv`
> 
> ### Issue
> I'm able to see the resource if I filter on group, which is expected.
> 
> ```shell
> $ rad resource list Applications.Core/containers -a todoapp -g peachgroup
> RESOURCE       TYPE                          GROUP       ENVIRONMENT  STATE
> todocontainer  Applications.Core/containers  peachgroup  default      Succeeded
> ```
> 
> If I add environment I cannot see the resource.
> 
> ```shell
> $ rad resource list Applications.Core/containers -a todoapp -e peachenv -g peachgroup
> RESOURCE  TYPE      GROUP     ENVIRONMENT  STATE
> ```
> 
> If I filter on environment only, I cannot see the resources. The result looks like this:
> 
> ```shell
> $ rad resource list Applications.Core/containers -a todoapp -e peachenv
> The application "todoapp" could not be found in workspace "k3d". Make sure you specify the correct application with '-a/--application'
> ```

@brooke-hamilton  I've taken a look at this, and found some issues with how the filtering was working. I have pushed up a change, and set up the following environment:

1. Checkout the PR and run make install to install the rad CLI.
2. Use a new k3d cluster, and run `rad init` on the cluster.
3. Create a new group called ``redgroup``
4. Create a new environment called ``redenv``
5. Deploy ``app.bicep`` to the ``redenv``
6. Run the following commands with different flags.

```bash
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad group switch redgroup
Switched to resource group "redgroup"
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad env list
RESOURCE  TYPE                            GROUP     ENVIRONMENT  STATE
redenv    Applications.Core/environments  redgroup  default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -e redenv
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -g redgroup
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -e redenv
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -g redgroup
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -g redgroup -e redenv
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  redgroup  redenv       Succeeded
```

I can confirm that the `rad resource list` is working as expected in the `redenv` environment. I then switch to the `default` group (where I have deployed `app.bicep` to), and run the following commands:

```bash
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad group switch default
Switched to resource group "default"
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -e default
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -g default
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -e default
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -g default
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
vscode ➜ /workspaces/radius (feature/7789) $ ./dist/linux_arm64/release/rad resource list Applications.Core/containers -a radius -g default -e default
RESOURCE  TYPE                          GROUP     ENVIRONMENT  STATE
demo      Applications.Core/containers  default   default      Succeeded
```

I believe that this solves the issues that you were seeing. Any feedback or further concerns, let me know.

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Merge branch 'main' into feature/7789

willvelida committed:
Merge branch 'main' into feature/7789

brooke-hamilton committed:
Merge branch 'main' into feature/7789

brooke-hamilton reviewed (changes requested):
This is getting very close to finished. There is a functional test that is failing. I can help track that down. Also, let's get aligned on the test scenarios. I think we are close. I'll reach out to you for a conversation on that. Thanks again for the help with this feature. 🥇
